From f02f156a66ef993edf8bd66794f4e9706d403cc7 Mon Sep 17 00:00:00 2001
From: Felix Kuehling <Felix.Kuehling@amd.com>
Date: Sat, 9 Apr 2016 18:12:18 -0400
Subject: [PATCH 1763/2094] drm/amdgpu: Report FB as private if BAR is above
 40bit

When the BAR is above 40bit, the framebuffer is not peer-accessible
(although it is still CPU accessible). For HSA purposes, report
FB memory as private in this case.

Change-Id: Ia8d5d5d7e258d5008daab13b3992c9065c712ec0
Signed-off-by: Felix Kuehling <Felix.Kuehling@amd.com>
---
 drivers/gpu/drm/amd/amdgpu/amdgpu_amdkfd.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/gpu/drm/amd/amdgpu/amdgpu_amdkfd.c b/drivers/gpu/drm/amd/amdgpu/amdgpu_amdkfd.c
index 50476e9..f7c2a5a 100644
--- a/drivers/gpu/drm/amd/amdgpu/amdgpu_amdkfd.c
+++ b/drivers/gpu/drm/amd/amdgpu/amdgpu_amdkfd.c
@@ -347,11 +347,11 @@ void get_local_mem_info(struct kgd_dev *kgd,
 		mem_info->local_mem_size_public = rdev->mc.visible_vram_size;
 		mem_info->local_mem_size_private = rdev->mc.real_vram_size -
 				rdev->mc.visible_vram_size;
-		mem_info->vram_width = rdev->mc.vram_width;
 	} else {
-		pr_err("amdgpu: vram aperture is out of 40bit address base: 0x%llx limit 0x%llx\n",
-				rdev->mc.aper_base, aper_limit);
+		mem_info->local_mem_size_public = 0;
+		mem_info->local_mem_size_private = rdev->mc.real_vram_size;
 	}
+	mem_info->vram_width = rdev->mc.vram_width;
 
 	pr_debug("amdgpu: address base: 0x%llx limit 0x%llx public 0x%llx private 0x%llx\n",
 			rdev->mc.aper_base, aper_limit,
-- 
2.7.4

